pkg/test/framework/components/registryredirector/kube.go

```diff
@@ -87,6 +96,12 @@ func newKube(ctx resource.Context, cfg Config) (Instance, error) {
 		return nil, fmt.Errorf("failed to apply rendered %s, err: %v", env.RegistryRedirectorServerInstallFilePath, err)
 	}
 
+	fetchFn := testKube.NewPodFetch(ctx.Clusters().Default(), c.ns.Name(), podSelector, fmt.Sprintf("uuid=%s", instanceID))
```

Comment:
do we need the uuid selector?

Reply:
Sometimes, when there are two or more instances are existing in the different namespace, Service seems to forward traffics to the unintended instances. I did not exactly check such behaviors in detail, yet.